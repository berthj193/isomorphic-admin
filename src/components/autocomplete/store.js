import React from 'react';
import PropTypes from 'prop-types';
import withStoresManager from '../apolloProviders/withStoresManager';
import storeQueries from '../../graphql/stores';
import Select, { SelectOption } from '../uielements/select';

export class StoreAutocomplete extends React.Component {
  static propTypes = {
    stores: PropTypes.arrayOf(PropTypes.shape({
      key: PropTypes.string,
    })),
  };

  static defaultProps = {
    stores: [],
  };

  dataSource = (this.props.stores || [])
    .map(({ key }) =>
      <SelectOption value={key} key={key}>{`${key}`}</SelectOption>
    );

  render() {
    return (
      <Select
        showSearch
        {...this.props}
      >
        {this.dataSource}
      </Select>
    );
  }
}

export default withStoresManager(StoreAutocomplete, [
  {
    query: storeQueries.storesAutocomplete,
  },
]);
