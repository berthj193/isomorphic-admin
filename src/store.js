import { createStore, applyMiddleware, compose } from 'redux';
import createHistory from 'history/createBrowserHistory';
import { routerMiddleware } from 'react-router-redux';
import thunk from 'redux-thunk';
import createSagaMiddleware from 'redux-saga';
import reducers from './reducers';
import rootSaga from './sagas';

const history = createHistory();
const sagaMiddleware = createSagaMiddleware();
const routeMiddleware = routerMiddleware(history);
const middlewares = [thunk, sagaMiddleware, routeMiddleware];
const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

const store = createStore(
  reducers,
  composeEnhancers(applyMiddleware(...middlewares))
);
sagaMiddleware.run(rootSaga);
export { store, history };
