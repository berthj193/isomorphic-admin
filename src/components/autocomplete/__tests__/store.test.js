import React from 'react';
import { StoreAutocomplete } from '../store';
import { shallow } from 'enzyme';

describe('stores autocomplete component', () => {
  const getComponent = (props = {}) => shallow(<StoreAutocomplete {...props} />);
  const dataObj = {
    id: '1',
    key: 'foo',
  };

  it('datasource should return empty array when there are no elements', () => {
    const instance = getComponent().instance();
    expect(instance.dataSource.length).toBe(0);
  });

  it('datasource should return one element', () => {
    const instance = getComponent({
      stores: [dataObj],
    }).instance();

    expect(instance.dataSource).toMatchSnapshot();
  });
});
