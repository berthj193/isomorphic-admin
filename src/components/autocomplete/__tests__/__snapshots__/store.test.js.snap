// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`stores autocomplete component datasource should return one element 1`] = `
Array [
  <Option
    value="foo"
>
    foo
</Option>,
]
`;
